torchinductor/codegen/triton_template.py

```diff
@@ -120,45 +180,108 @@ def call_kernel(self, wrapper, name: str):
         # def grid(META):
         #     return (...)
         # kernel1[grid](arg0, arg1, ...)
-        extra_arg_defs, extra_call_args = self.args.python_argdefs()
-        extra_args = ", ".join(extra_call_args)
-        self_args = ", ".join({**self.args_dict, **self.const_dict}.values())
+        extra_args = ", ".join(self.extra_call_args)
+        self_args = ", ".join({**self.inout_dict, **self.args_dict}.values())
+        self_const_kwargs = ", ".join(f"{k}={v}" for k, v in self.const_dict.items())
         args = self_args + (
             ", " + extra_args if extra_args and len(extra_args) > 0 else ""
         )
+        args_kwargs = args + ", " + self_const_kwargs
         wrapper.writeline(self.gen_grid(name))
-        wrapper.writeline(f"{name}[grid_{name}]({args})")
+        wrapper.writeline(f"{name}[grid_{name}]({args_kwargs})")
 
 
-def template_codegen(scheduler, node):
+def template_codegen(scheduler, scheduler_node):
     """
     codegen function for triton templates
     scheduler: Scheduler
-    node: ExternKernelSchedulerNode
+    scheduler_node: ExternKernelSchedulerNode
     """
     wrapper = V.graph.wrapper_code
-    deivce, group = node.group
+    deivce, group = scheduler_node.group
 
     reschedule = []
-    with scheduler.kernel(TritonTemplateKernel(node.node, *group)) as kernel:
+    fuse = False
+    could_remove_kernel_buf = False
+    fusable_nodes = []
+    with scheduler.kernel(TritonTemplateKernel(scheduler_node.node, *group)) as kernel:
+        # map const args/ shape/ strides to kernel args
+        kernel.map_args()
+        # set self.args name to match the TritonTemplateKernel's args names
+        kernel.rename_vars()
+        # update node dep from StarDep to MemoryDep
+        # node.update_dep_type()
+        # mark node of TritonTemplateKernel as fusable and update fusable_deps
+        scheduler_node.mark_fusable()
+        # enqueue any nodes that became runable after this node is run
+        # otherwise, relu after conv is in blocked_nodes that could not be in
+        # runable_groups to be fused to conv
+        scheduler.barrier()
```

Comment:
I think this isn't exactly correct because we aren't checking the order elements are read in.  We should make `mark_fusable()` generate the correct MemoryDep.

Reply:
Use `scheduler_node.update_mem_dep()` instead